src/components/GoalsComponents/GoalConfigModal/ConfigGoal.tsx

```diff
@@ -102,20 +102,30 @@ const ConfigGoal = ({ goal, action }: { action: "Update" | "Create"; goal: GoalI
     return title.length === 0 || title.trim() === "";
   };
 
-  const getFinalTags = () => ({
-    ...goal,
-    // due: due && due !== "" ? new Date(due).toString() : null,
-    // start: start && start !== "" ? new Date(start).toString() : null,
-    duration: tags.duration !== "" ? `${tags.duration}` : null,
-    afterTime,
-    beforeTime,
-    habit: tags.repeatWeekly ? "weekly" : null,
-    on: calDays.filter((ele) => tags.on.includes(ele)),
-    timeBudget: {
-      perDay: state.goalType === "Budget" ? perDayHrs.join("-") : null,
-      perWeek: state.goalType === "Budget" ? perWeekHrs.join("-") : null,
-    },
-  });
+  const getFinalTags = () => {
```

Comment:
Hi @vinaybadgujar102 please revert to old code. as this approach is just increasing the loc(s). To solve the issue just use the ternary operator ( for afterTime & beforeTime ) i.e. if its a budget goal put the timings otherwise set it to null